### PR TITLE
use form validation in `TextFilter`

### LINKED ISSUE #2200: Table: Disable filter button bar action button on no input
### Describe the bug (current behavior)

The action button in the filter pop up is enabled when no filter value is input.

image

### Expected Behavior

Suggest to disable the action button when there is no valid input in the filter pop up.

### Link to minimal repro

https://stackblitz.com/edit/github-lpys6q-xwznam?file=src%2FmyTable.tsx

### Steps To Reproduce

Add any built-in filter type to a table and open the filter. The action button is enabled by default.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Fixes #2200

1. `BaseFilter` now renders a `<form>` element.
2. `TextFilter` now renders `<input required>` for built-in browser validation.
3. `FilterButtonBar` now renders `<button type="submit">`.

Together, this prevents the filter from submitting when the text input is empty.

**Note**: `<form>` comes with built-in keyboard support ([`submit`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event) event is dispatched when <kbd>enter</kbd> is pressed), so I was able to remove explicit `keydown` handlers and related tests.

## Testing

Confirmed that browser prevents submitting TextFilter with an empty value.

Also confirmed that all filters in the [Filters story](https://itwin.github.io/iTwinUI/2205/react/?story=table--filters&theme=dark) continue to work ok, both with mouse and keyboard.

## Docs

Added changesets.

## After-PR TODOs

- [ ] Fix datepicker popover (see [comment](https://github.com/iTwin/iTwinUI/pull/2205#discussion_r1731714448)).

### CODE REVIEW

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx [hunk 1]
```diff
@@ -114,26 +114,15 @@ export const DateRangeFilter = <T extends Record<string, unknown>>(
     });
   }, []);
 
-  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (smmr-dn) [inline packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx:117]: Just out of curiosity, why did we remove the user's ability to submit filters by pressing Enter? Could we still allow `onKeyDown` and call `onSubmit`?

File: packages/itwinui-react/src/core/Table/filters/FilterButtonBar.tsx [hunk 2]
```diff
@@ -60,7 +62,7 @@ export const FilterButtonBar = (props: FilterButtonBarProps) => {
   return (
     <Box className={cx('iui-button-bar', className)} style={style} id={id}>
       {children}
-      <Button styleType='high-visibility' onClick={setFilter}>
+      <Button type='submit' styleType='high-visibility' onClick={setFilter}>
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Table/filters/FilterButtonBar.tsx:65]: Yes, if both props are being passed, then both will be called. The existing `setFilter` code should work the same as before when `onSubmit` is not passed.

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx [hunk 3]
```diff
@@ -114,26 +114,15 @@ export const DateRangeFilter = <T extends Record<string, unknown>>(
...
-
-    if (event.key === 'Enter') {
-      setFilter([from, to]);
-    }
-  };
-
   return (
-    <BaseFilter>
+    <BaseFilter onSubmit={() => setFilter([from, to])}>
       <DatePickerInput
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx:119]: Ah, this must be a regression from recent Popover changes (e.g. #2185). Will look into it afterwards.

File: packages/itwinui-react/src/core/Table/filters/TextFilter/TextFilter.test.tsx [hunk 4]
```diff
@@ -37,21 +37,6 @@ it('should call setFilter with input value', () => {
   expect(setFilter).toHaveBeenCalledWith('test filter');
 });
 
-it('should call setFilter when Enter is pressed', () => {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Table/filters/TextFilter/TextFilter.test.tsx:40]: I don't think it's necessary. Previously we were testing the code that we had written, whereas now we're just relying on the default browser behavior.

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx:117]: We haven't removed any functionality. We are now using a `<form>` which automatically gets submitted when the user presses <kbd>Enter</kbd> from anywhere inside it.

Comment (smmr-dn) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DateRangeFilter.tsx:117]: I see, that makes sense, `<form>` must do that automatically already.

**Pull request merged**